package/tests/test_server.py

```diff
@@ -243,6 +255,79 @@ def test_nodes_endpoint(cli_runner, client):
     assert data == EXPECTED_PIPELINE_DATA
 
 
+@pytest.mark.usefixtures("patched_get_project_context")
+def test_pipeline_flag(cli_runner, client):
+    """Test that running viz with `--pipeline` flag will return a correct pipeline"""
+    cli_runner.invoke(server.commands, ["viz", "--pipeline", "another"])
```

Comment:
Could you add a test for trying to load a pipeline that doesn't exist?

Reply:
Good point! I will add an extra test and also catch the KeyError in `except`.